### PR TITLE
Make maximum amount of spans configurable

### PR DESCRIPTION
**Pull request by fsrv-xyz**
In one of my projects, I need to do performance monitoring for complex tasks with the highest possible accuracy.
Due to some of the tasks are long running, the current `const` span limit of 1000 per transaction will be not enough.

I saw, that some other SDKs (for example python) have an option for that.

### CODE REVIEW

**Comment by YaroslavPodorvanov** [comment]
I think exists config on server too and user can not change config on server

**Comment by fsrv-xyz** [comment]
I have tested on one of our self hosted instances. Works fine with a higher limit (as far as I can check).

**Comment by cleptric** [comment]
@fsrv-xyz Can you add some tests for this so we get it merged?

**Comment by fsrv-xyz** [comment]
@cleptric Sure. I added some test cases covering the added code (and some additional things).

File: span_recorder.go [hunk 1]
```diff
@@ -20,6 +15,10 @@ type spanRecorder struct {
 // record stores a span. The first stored span is assumed to be the root of a
 // span tree.
 func (r *spanRecorder) record(s *Span) {
+	maxSpans := defaultMaxSpans
+	if CurrentHub().Client() != nil {
+		maxSpans = CurrentHub().Client().options.MaxSpans
```
Comment (cleptric) [inline span_recorder.go:20]:
```suggestion
	if client := CurrentHub().Client(); client != nil {
		maxSpans = client.Options().MaxSpans
```

File: client.go [hunk 2]
```diff
@@ -175,6 +180,8 @@ type ClientOptions struct {
 	Environment string
 	// Maximum number of breadcrumbs.
 	MaxBreadcrumbs int
+	// Maximum number of spans.
+	MaxSpans int
```
Comment (cleptric) [inline client.go:184]:
```suggestion
	// Maximum number of spans.
	//
	// See https://develop.sentry.dev/sdk/envelopes/#size-limits for size limits
	// applied during Event ingestion. Events that exceed these limits might get dropped.
	MaxSpans int
```

**Review by cleptric — COMMENTED** [review]
Please update the `CHANGELOG.md` as well and add something like 

> feat: Make maximum amount of spans configurable (#460)

Thanks for your work on this 🙂

File: span_recorder_test.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+			Logger.SetOutput(&logBuffer)
+			defer Logger.SetOutput(io.Discard)
+			spanRecorder := spanRecorder{}
+
+			currentHub.BindClient(&Client{
+				options: ClientOptions{
+					MaxSpans: tt.maxSpans,
+				},
+			})
+			// unbind client after test for not affecting other tests
```
Comment (cleptric) [inline span_recorder_test.go:44]:
```suggestion
			// Unbind the client afterwards, to not affect other tests
```

**Comment by fsrv-xyz** [comment]
I applied the changes you suggested and updated the changelog file.

**Review by cleptric — APPROVED** [review]

**Pull request merged**